### PR TITLE
Make Credential a proper data type

### PR DESCRIPTION
**Pull request by UlfNorell**
This simplifies the marshalling to Haskell since we want a data type on the Haskell side and not an `Either`. Also it makes the spec easier to follow.

### CODE REVIEW

File: src/Ledger/Transaction.lagda [hunk 1]
```diff
@@ -66,7 +66,7 @@ the transaction body are:
   field crypto : _
   open Crypto crypto public
   open Ledger.TokenAlgebra ScriptHash public
-  open Ledger.Address Network KeyHash ScriptHash public
+  open Ledger.Address Network KeyHash ScriptHash ⦃ it ⦄ ⦃ it ⦄ ⦃ it ⦄ public
```
Comment (WhatisRT) [inline src/Ledger/Transaction.lagda:69]: Is there a reason you added these? It works for me without them. Does that break with the new instance resolution?

Comment (UlfNorell) [reply src/Ledger/Transaction.lagda:69]: It broke some stuff for Haskell type generation.

Module applications don't add implicit/instance argument at the end, so without this, the types you get all take instance arguments that are resolved at use site, so when you want to generate a Haskell type for `Credential` say, the tactic needs to deal with types parameterised by instance arguments.

Comment (WhatisRT) [reply src/Ledger/Transaction.lagda:69]: Ah, makes sense. Thank you!

**Review by WhatisRT — APPROVED** [review]
LGTM

**Pull request merged**